app/src/main/java/eu/kanade/tachiyomi/ui/setting/SettingsRelatedController.kt

```diff
@@ -0,0 +1,194 @@
+                chooseFile.setType("*/*")
+                val intent = Intent.createChooser(chooseFile, "Choose a file")
+                try {
+                    startActivityForResult(intent, RELATED_FILE_PATH_L)
+                } catch (e: ActivityNotFoundException) {
+                    startActivityForResult(preferences.context.getFilePicker("/"), RELATED_FILE_PATH_L)
+                }
+            }
+
+            preferences.relatedLastUpdated().asObservable()
+                    .subscribeUntilDestroy { path ->
+                        summary = path
+                    }
+        }
+
+        preference {
+            title = "Github"
+            val url = "https://github.com/goldbattle/MangadexRecomendations"
+            summary = url
+            onClick {
+                val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
+                startActivity(intent)
+            }
+            isIconSpaceReserved = true
+        }
+
+    }
+
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        when (requestCode) {
+            RELATED_FILE_PATH_L -> if (data != null && resultCode == Activity.RESULT_OK) {
+
+                //=============================================================
+                //=============================================================
+
+                // Parse the url from the specified uri
+                val context = applicationContext ?: return
+                val activity = activity ?: return
+                val selectedFile = data.data
+                val input: InputStream? = context.getContentResolver()?.openInputStream(data.data!!)
+
+                // Check if we where able to open it
+                if(input==null) {
+                    context.toast("Unable to open file $selectedFile")
+                    return
+                }
+
+                //=============================================================
+                //=============================================================
+                
+
+                // Load everything from the buffer into our string
+                val result = input.bufferedReader().use { it.readText() }
+                val relatedPageResult: JSONObject
+                try {
+                    relatedPageResult = JSONObject(result)
+                } catch (e : JSONException) {
+                    context.toast(context.resources.getString(R.string.pref_related_broken_file))
+                    return
+                }
+
+                // Count the number of mangas we need to load
+                var totaMangas: Int = 0
+                for(key in relatedPageResult.keys()) {
+                    totaMangas++
+                }
+
+                // Display an alert to the user (replaces the ProgressDialog)
+                // https://stackoverflow.com/a/51694457
+                val builder = AlertDialog.Builder(activity)
+                builder.setCancelable(false)
```

Comment:
It shouldn't lock the app since it is just a notification now.
Right now it exits if people force close the app, but it shouldn't impact the usage now.